src/modules/order/order.service.js

```
@@ -33,39 +33,52 @@ const getAllOrders = async (status, user, page = 1, limit = 10) => {
+    query.user = user.id; // Use the ID directly from user object
+    logger.debug(`getAllOrders: Customer role - filtering by user ID: ${user.id}`);
   } else {
-    logger.debug('getAllOrders: Admin user - showing all orders');
+    // Admin - see all orders
+    logger.debug('getAllOrders: Admin role - showing all orders');
   }
   
   try {
-    // Convert page and limit to numbers with defaults
+    // Convert pagination parameters
     const pageNum = parseInt(page) || 1;
```

Biome (1.9.4)

[error] Use Number.parseInt instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseInt instead.


(lint/style/useNumberNamespace)